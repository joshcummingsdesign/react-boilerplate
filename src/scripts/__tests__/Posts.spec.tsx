import * as React from 'react';
import { shallow } from 'enzyme';
import Posts from '../components/Posts';
import PostCard from '../components/PostCard';
const posts = require('../__mocks__/posts.json');

test('Posts renders correctly', () => {
  const component = shallow(<Posts posts={posts} searchTerm="" />);
  expect(component).toMatchSnapshot();
});

test('Posts should render correct amount of shows', () => {
  const component = shallow(<Posts posts={posts} searchTerm="" />);
  expect(component.find(PostCard).length).toEqual(posts.length);
});
