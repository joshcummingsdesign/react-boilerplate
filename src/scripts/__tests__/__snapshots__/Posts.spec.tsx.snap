// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts renders correctly 1`] = `
<div
  className="posts"
>
  <h1>
    You Searched for 
  </h1>
  <PostCard
    key="1"
    post={
      Object {
        "body": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
        "id": 1,
        "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
        "userId": 1,
      }
    }
  />
  <PostCard
    key="2"
    post={
      Object {
        "body": "est rerum tempore vitae
sequi sint nihil reprehenderit dolor beatae ea dolores neque
fugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis
qui aperiam non debitis possimus qui neque nisi nulla",
        "id": 2,
        "title": "qui est esse",
        "userId": 1,
      }
    }
  />
  <PostCard
    key="3"
    post={
      Object {
        "body": "et iusto sed quo iure
voluptatem occaecati omnis eligendi aut ad
voluptatem doloribus vel accusantium quis pariatur
molestiae porro eius odio et labore et velit aut",
        "id": 3,
        "title": "ea molestias quasi exercitationem repellat qui ipsa sit aut",
        "userId": 1,
      }
    }
  />
</div>
`;
